src/firefly/js/visualize/DrawLayerCntlr.js

```diff
@@ -388,6 +397,27 @@ function determineAndCallLayerReducer(state,action,dlReducer,force) {
 }
 
 
+function clearPreattachLayer(state,action) {
+    var {drawLayerId}= action.payload;
+    var drawLayer= state.drawLayerAry.find( (dl) => drawLayerId===dl.drawLayerId);
+    if (drawLayer) return state;
+    if (!state.preAttachedTypes[drawLayer.drawLayerTypeId]) return state;
+
+    const preAttachedTypes= clone(preAttachedTypes);
+    Reflect.deleteProperty(preAttachedTypes, drawLayerTypeId);
+
+    return clone(state, {preAttachedTypes});
```

Comment:
An alternative for removing object props within a reducer is to use react update + lodash omit.
i.e  update(obj, {key: {$set: omit(obj, 'prop')})